### PR TITLE
BoP: Address new branch and target namings

### PR DESCRIPTION
**Pull request by petr-muller**
This commit follows the newly agreed (and probably) final decisions on staging branch structure, names and build target mappings.

The changes are summarized as follows:
1.  There will be just one staging branch per major product release, and it will be _prefixed_ with staging- rather than suffixed. Example: new staging branches are called _staging-rhel-7_ instead of earlier _rhel-7.2-staging_.
2. CI does not attempt to map a branch to an appropriate target anymore, instead a wildcard target _staging-rhel-7-candidate_ will be used, and RCM will take the responsibility for making sure this target always uses latest build tag available.

The jobs using this candidate code were deployed to stage:
1. _Component setup:_ 'platform-component-setup-testing'
2. _Commit dispatcher:_ 'ci-setup-dispatcher-commit', build 43 was performed using the candidate code and PASSed.
3. _BoP worker:_ 'ci-setup-commit-staging-rhel-7', build 1 was triggered by the dispatcher run above and correctly scheduled a build against staging-rhel-7-candidate target.

In short: everything seems to work as expected :)

### CODE REVIEW

**Comment by petr-muller** [comment]
@lulinqing , @gsr-shanks looks good?

File: build-on-push/platform_ci/platform_ci/distgit_test.py [hunk 1]
```diff
@@ -17,20 +17,22 @@
...
 
 
 class DistGitBranchTest(unittest.TestCase):
-    STAGING = {"branch": "rhel-7.1-staging", "target": "rhel-7.1-candidate"}
+    STAGING = {"branch": "staging-rhel-7", "target": "staging-rhel-7-candidate"}
     PRIVATE = {"branch": "private-pmuller-branch", "target": None}
-    STANDARD = {"branch": "extras-rhel-7.2", "target": "extras-rhel-7.2-candidate"}
-    EXTRAS_STAGING = {"branch": "extras-rhel-7.2-staging", "target": "extras-rhel-7.2-candidate"}
+    STANDARD = {"branch": "rhel-7.2", "target": "rhel-7.2-candidate"}
+    EXTRAS_STANDARD = {"branch": "extras-rhel-7.2", "target": "extras-rhel-7.2-candidate"}
```
Comment (lulinqing) [inline build-on-push/platform_ci/platform_ci/distgit_test.py:27]: Should the targets here be "rhel-7-candidate"/"extras-rhel-7-candidate"?

**Comment by gsr-shanks** [comment]
I vaguely recollect that we will be using brew 'tags' and not 'targets'. Did anything change in that regard ?

File: build-on-push/platform_ci/platform_ci/distgit_test.py (hunk 1 above)
Comment (petr-muller) [inline build-on-push/platform_ci/platform_ci/distgit_test.py:27]: As long as the underlying branch is tied to a minor-version of a product (like all standard branches: we have "rhel-7.2" branch, "extras-rhel-7.2" branch etc) they should be build against a matching target: it does not make sense to build  "rhel-7.2" branch against "rhel-7-candidate" because that one is likely to point to a tag inheriting rhel-7.3-candidate.

Currently, BoP does not auto-build standard branches anyway, so it does not matter much.

**Comment by petr-muller** [comment]
@gsr-shanks in different context, I believe. BoP is all about automated builds whenever we have new code in DistGit. For issuing builds, you need a build target. The logic here is dealing mostly about which target to use for building which branch.

The tags should be then used when _testing_ builds, that is for identification of the product version for which the build is intended. There using targets is misleading because 1) targets can change their destination tags and 2) it is not suitable for various single-purpose targets that we use, but which point to a standard destination tag that we would be able to process.

**Comment by petr-muller** [comment]
I hope I managed to explain both matters, so I'll be bold and merge this PR. If there are further concerns/follow-ups, feel free to contact me, file an issue or anything.
